geo-types/src/polygon.rs

```diff
@@ -468,3 +468,43 @@ impl<T: CoordinateType> From<Rect<T>> for Polygon<T> {
         )
     }
 }
+
+#[cfg(feature = "rstar")]
+impl<T> ::rstar::RTreeObject for Polygon<T>
+where
+    T: ::num_traits::Float + ::rstar::RTreeNum,
+{
+    type Envelope = ::rstar::AABB<Point<T>>;
+
+    fn envelope(&self) -> Self::Envelope {
+        let mut minx = ::num_traits::Float::max_value();
+        let mut maxx = ::num_traits::Float::min_value();
+        let mut miny = ::num_traits::Float::max_value();
+        let mut maxy = ::num_traits::Float::min_value();
+        for p in &self.exterior.0 {
+            if p.x < minx {
+                minx = p.x;
+            }
+            if p.x > maxx {
+                maxx = p.x;
+            }
+            if p.y < miny {
+                miny = p.y;
+            }
+            if p.y > maxy {
+                maxy = p.y;
+            }
+        }
+        ::rstar::AABB::from_corners(Point::new(minx, miny), Point::new(maxx, maxy))
+    }
```

Comment:
is this implementation different than doing this?

```rust
#[cfg(feature = "rstar")]
impl<T> ::rstar::RTreeObject for Polygon<T>
where
    T: ::num_traits::Float + ::rstar::RTreeNum,
{
    type Envelope = ::rstar::AABB<Point<T>>;

    fn envelope(&self) -> Self::Envelope {
        self.exterior.envelope()
    }
}
```

which would [reuse this implementation](https://github.com/georust/geo/blob/be288752844bfa96220cf030a985c0b6487c7c44/geo-types/src/line_string.rs#L192-L213)

Reply:
It would create the same envelope. Didn't know that it was implemented for LineString already.